content/3_Silver/More_Prefix_Sums.mdx

```diff
@@ -652,6 +652,85 @@ for _ in range(query_num):
 
 <Problems problems="cum2" hideSuggestProblemButton />
 
+## Difference Arrays
+
+<Resources>
+	<Resource source="Codeforces" title="An Introduction To Difference Arrays" url="https://codeforces.com/blog/entry/78762"/>
+</Resources>
+
+<FocusProblem problem="sample3" />
+
+### Explanation
+
+Let's create an array $s$, where $s[i]$ is the number of times operation $i$ is applied.
+The important step is how we update it.
+
+For an interval $[l, r]$, we can't loop through the interval and
+increment each value, as that would be $\mathcal{O}(MK)$ and too slow.
+Instead, we increment $s[l]$ by one and decrement $s[r+1]$ by one.
+
+Now, we get the *actual* array by computing its prefix sum array,
+resulting in $\mathcal{O}(M)$ time complexity.
+The second part, applying the operations, can be done exactly the same way.
+
+### Implementation
+
+**Time Complexity:** $\mathcal{O}(N+M)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <array>
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+int main() {
+	int n, m, k;
+	cin >> n >> m >> k;
+	vector<int> a(n + 1);
```

Comment:
is 1-indexing needed

Reply:
its always way more convenient for prefix sum problems

Reply:
For convenience yes